### PR TITLE
Issue #153: Make sure the menu module is enabled before using menu_*() functions.

### LINKED ISSUE #153: Dependency missing for menu_*() functions
As the `registryonsteroids_alter` module using `menu_get_menus()` function and other `menu_*()` functions, there should be added `menu` dependency as the very minimal drupal instance with enabled of `registryonsteroids_alter` returns error like this:

`Fatal error: Call to undefined function menu_get_menus() in /sites/all/modules/contrib/registryonsteroids/modules/registryonsteroids_alter/registryonsteroids_alter.alter.inc on line 114`

### PR DESCRIPTION
**Pull request by drupol**
Fix #153

### CODE REVIEW

**Comment by donquixote** [comment]
I have not thought in depth about possible implications, but it seems alright.

Hint for reviewers:
```
git clone git@github.com:drupol/registryonsteroids.git
cd registryonsteroids
git show 9ae251a --word-diff
```
This shows the actual change, without noise from indentation.

**Comment by donquixote** [comment]
I just learned about the ?w=1 option!
https://github.com/drupol/registryonsteroids/pull/154/files?w=1

File: modules/registryonsteroids_alter/registryonsteroids_alter.alter.inc [hunk 1]
```diff
@@ -110,22 +110,24 @@ function registryonsteroids_alter_block_view_alter(&$data, $block) {
...
-      $content = registryonsteroids_menu_tree(
-        $block->delta,
-        [
-          'block' => $block,
-          'suggestions parts' => [],
-        ]
-      );
-      $content['#contextual_links'] = $data['content']['#contextual_links'];
-      $data['content'] = $content;
+  if (module_exists('menu')) {
```
Comment (donquixote) [inline modules/registryonsteroids_alter/registryonsteroids_alter.alter.inc]: Looking closer at this, I have to disagree with the PR.

There are menu blocks provided by system module (the built-in default menus) and other menu blocks provided by menu module (user-created menus). The 'menu' module should only be required for the latter ones.

Background: What is the purpose of the 'menu' module in Drupal 7?

The core (without menu module):
- creates a 'menu_links' table.
- fills the menu_links table (and the menu_router table) based on hook_menu() information.
- provides CRUD functions for menu links, like menu_link_save().
- creates some default menus (not sure exactly how that works).
- registers blocks for the default menus, see system_block_info().
- It uses menu_list_system_menus() to get a list of all system menus.

The menu module:
- provides a database structure to create new custom menus (the 'menu_custom' table).
- provides a UI to create new custom menus.
- provides a UI to create and modify menu links for system menus and custom menus.
- Registers additional blocks for the newly added menus, see menu_block_info().
- It uses menu_get_menus(FALSE) to get a list of all custom, user-created menus.

Comment (drupol) [reply modules/registryonsteroids_alter/registryonsteroids_alter.alter.inc]: Very nice catch. Will update this now.

**Comment by drupol** [comment]
Merging this to move forward with further optimizations.

**Pull request merged**